<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MicroplotGeneratorDialogBase</class>
 <widget class="QDialog" name="MicroplotGeneratorDialogBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>450</width>
    <height>750</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Create Microplots Grid</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="grid_params_group">
     <property name="title">
      <string>Grid Parameters</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Number of Rows:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="rows_spinbox"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Number of Columns:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="cols_spinbox"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="plot_size_group">
     <property name="title">
      <string>Plot Size</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Plot Length (m):</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="length_spinbox"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Plot Width (m):</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QDoubleSpinBox" name="width_spinbox"/>
      </item>
      </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="numbering_group">
     <property name="title">
      <string>Plot Numbering</string>
     </property>
     <layout class="QFormLayout" name="formLayout_4">
      <item row="0" column="0">
       <widget class="QLabel" name="label_14">
        <property name="text">
         <string>ID Format:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="id_format_combobox">
        <item>
         <property name="text">
          <string>Sequential</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Serpentine</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_15">
        <property name="text">
         <string>Start Numbering From:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="start_numbering_combobox">
        <item>
         <property name="text">
          <string>Bottom Left</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Bottom Right</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Top Left</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Top Right</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_16">
        <property name="text">
         <string>Initial ID (optional):</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="initial_id_spinbox">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1000000</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="instructions_label">
        <property name="text">
         <string>Click on the map to select the 4 corner points in the following order: Top Left, Top Right, Bottom Right, Bottom Left.</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Point 1 (Top Left):</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="point1_x"/>
        </item>
        <item row="0" column="2">
         <widget class="QLineEdit" name="point1_y"/>
        </item>
        <item row="0" column="3">
         <widget class="QPushButton" name="select_point1_button">
          <property name="text">
           <string>Select</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Point 2 (Top Right):</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="point2_x"/>
        </item>
        <item row="1" column="2">
         <widget class="QLineEdit" name="point2_y"/>
        </item>
        <item row="1" column="3">
         <widget class="QPushButton" name="select_point2_button">
          <property name="text">
           <string>Select</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Point 3 (Bottom Right):</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="point3_x"/>
        </item>
        <item row="2" column="2">
         <widget class="QLineEdit" name="point3_y"/>
        </item>
        <item row="2" column="3">
         <widget class="QPushButton" name="select_point3_button">
          <property name="text">
           <string>Select</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>Point 4 (Bottom Left):</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLineEdit" name="point4_x"/>
        </item>
        <item row="3" column="2">
         <widget class="QLineEdit" name="point4_y"/>
        </item>
        <item row="3" column="3">
         <widget class="QPushButton" name="select_point4_button">
          <property name="text">
           <string>Select</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="field_data_group">
     <property name="title">
      <string>Field Data (Optional)</string>
     </property>
     <layout class="QFormLayout" name="formLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="label_fieldmap_file">
        <property name="text">
         <string>Field Map File (.csv):</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QgsFileWidget" name="fieldmap_file_widget"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Plot Data File (.csv):</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QgsFileWidget" name="fielddata_file_widget"/>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Plot ID Join Column:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="fielddata_plotid_combobox"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="output_group">
     <property name="title">
      <string>Output</string>
     </property>
     <layout class="QFormLayout" name="formLayout_5">
      <item row="0" column="0">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>Output File (Optional):</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QgsFileWidget" name="output_file_widget"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="create_button">
     <property name="text">
      <string>Create Grid</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsFileWidget</class>
   <extends>QWidget</extends>
   <header>qgis.gui</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>